core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -258,12 +260,19 @@ public String analyzer() {
         return this.analyzer;
     }
 
+    @Deprecated
     public Boolean useAllFields() {
-        return useAllFields;
+        return fieldsAndWeights.size() == 1 && fieldsAndWeights.keySet().stream().anyMatch(Regex::isMatchAllPattern);
     }
 
+    /**
+     * This setting is deprecated, set {@link #field(String)} to "*" instead.
+     */
+    @Deprecated
     public SimpleQueryStringBuilder useAllFields(Boolean useAllFields) {
-        this.useAllFields = useAllFields;
+        if (useAllFields) {
+            this.fieldsAndWeights = Collections.singletonMap("*", 1.0f);
```

Comment:
this is technically breaking, because prior to this change the following would work:

```java
builder.field("foo", 1.5);
builder.useAllFields(true);
builder.useAllFields(false);
```

And since setting it to `true` permanently changes the `fieldsAndWeights`. While it seems contrived, I could see user code doing this passing the builder off to independent functions, the builder should be idempotent in this case